internal/harnesses/docker/docker.go

```diff
@@ -0,0 +1,127 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/docker/docker/api/types/mount"
+
+	"github.com/chainguard-dev/terraform-provider-imagetest/internal/containers/provider"
+	"github.com/chainguard-dev/terraform-provider-imagetest/internal/harnesses/base"
+	"github.com/chainguard-dev/terraform-provider-imagetest/internal/log"
+	"github.com/chainguard-dev/terraform-provider-imagetest/internal/types"
+)
+
+var _ types.Harness = &docker{}
+
+const DefaultDockerSocketPath = "/var/run/docker.sock"
+
+type docker struct {
+	*base.Base
+	id string
+
+	container provider.Provider
+}
+
+func New(id string, cli *provider.DockerClient, opts ...Option) (types.Harness, error) {
+	options := &HarnessDockerOptions{}
+	for _, opt := range opts {
+		if err := opt(options); err != nil {
+			return nil, err
+		}
+	}
+
+	dockerSocketPath := DefaultDockerSocketPath
+	if options.SocketPath != "" {
+		dockerSocketPath = options.SocketPath
+	}
+
+	var managedVolumes []mount.Mount
+	for _, vol := range options.ManagedVolumes {
+		managedVolumes = append(managedVolumes, mount.Mount{
+			Type:   mount.TypeVolume,
+			Source: vol.Source,
+			Target: vol.Destination,
+		})
+	}
+
+	var mounts []mount.Mount
+	mounts = append(mounts, mount.Mount{
+		Type:   mount.TypeBind,
+		Source: dockerSocketPath,
+		Target: DefaultDockerSocketPath,
+	})
+
+	for _, mt := range options.Mounts {
+		mounts = append(mounts, mount.Mount{
+			Type:   mount.TypeBind,
+			Source: mt.Source,
+			Target: mt.Destination,
+		})
+	}
+
+	container := provider.NewDocker(id, cli, provider.DockerRequest{
+		ContainerRequest: provider.ContainerRequest{
+			Ref:        options.ImageRef,
+			Entrypoint: []string{"/bin/sh", "-c"},
+			Cmd:        []string{"tail -f /dev/null"},
```

Comment:
```suggestion
			Cmd:        []string{"tail", "-f", "/dev/null"},
```

Any reason this isn't broken up like the entrypoint?

Reply:
No specific reason but all instances in the provider right now are like this. I don't see a compelling argument for splitting up either.

Reply:
https://github.com/chainguard-dev/terraform-provider-imagetest/blob/abd29064465d4085545cf810437c14172c6b9752/internal/harnesses/k3s/k3s.go#L57-L58

Reply:
I think the _correct_ way is the suggested, but given our entrypoint in this case it doesn't matter

Reply:
I can make it in a follow-up PR